File: packages/jsii-pacmak/test/expected.java/java/org/jsii/tests/calculator/SyncVirtualMethods.java
package org.jsii.tests.calculator;
@org.jsii.Jsii(module = org.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.SyncVirtualMethods")
public class SyncVirtualMethods extends org.jsii.JsiiObject {
    protected SyncVirtualMethods(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    public SyncVirtualMethods() {
        super(org.jsii.JsiiObject.InitializationMode.Jsii);
        org.jsii.JsiiEngine.getInstance().createNewObject(this);
    }
    public java.lang.Number callerIsMethod() {
        return this.jsiiCall("callerIsMethod", java.lang.Number.class);
    }
    public java.lang.Number callerIsAsync() {
        return this.jsiiAsyncCall("callerIsAsync", java.lang.Number.class);
    }
    public java.lang.Number virtualMethod(final java.lang.Number n) {
        return this.jsiiCall("virtualMethod", java.lang.Number.class, java.util.stream.Stream.of(n).toArray());
    }
    public void modifyValueOfTheProperty(final java.lang.String value) {
        this.jsiiCall("modifyValueOfTheProperty", Void.class, java.util.stream.Stream.of(value).toArray());
    }
    public java.lang.String retrieveValueOfTheProperty() {
        return this.jsiiCall("retrieveValueOfTheProperty", java.lang.String.class);
    }
    public java.lang.String retrieveReadOnlyProperty() {
        return this.jsiiCall("retrieveReadOnlyProperty", java.lang.String.class);
    }
    public void modifyOtherProperty(final java.lang.String value) {
        this.jsiiCall("modifyOtherProperty", Void.class, java.util.stream.Stream.of(value).toArray());
    }
    public java.lang.String retrieveOtherProperty() {
        return this.jsiiCall("retrieveOtherProperty", java.lang.String.class);
    }
    public java.lang.Number readA() {
        return this.jsiiCall("readA", java.lang.Number.class);
    }
    public void writeA(final java.lang.Number value) {
        this.jsiiCall("writeA", Void.class, java.util.stream.Stream.of(value).toArray());
    }
    public java.lang.Number getCallerIsProperty() {
        return this.jsiiGet("callerIsProperty", java.lang.Number.class);
    }
    public void setCallerIsProperty(final java.lang.Number value) {
        this.jsiiSet("callerIsProperty", value);
    }
    public java.lang.String getTheProperty() {
        return this.jsiiGet("theProperty", java.lang.String.class);
    }
    public void setTheProperty(final java.lang.String value) {
        this.jsiiSet("theProperty", value);
    }
    public java.lang.String getReadonlyProperty() {
        return this.jsiiGet("readonlyProperty", java.lang.String.class);
    }
    public java.lang.String getOtherProperty() {
        return this.jsiiGet("otherProperty", java.lang.String.class);
    }
    public void setOtherProperty(final java.lang.String value) {
        this.jsiiSet("otherProperty", value);
    }
    public java.lang.String getValueOfOtherProperty() {
        return this.jsiiGet("valueOfOtherProperty", java.lang.String.class);
    }
    public void setValueOfOtherProperty(final java.lang.String value) {
        this.jsiiSet("valueOfOtherProperty", value);
    }
    public java.lang.Number getA() {
        return this.jsiiGet("a", java.lang.Number.class);
    }
    public void setA(final java.lang.Number value) {
        this.jsiiSet("a", value);
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.Calculator/SyncVirtualMethods.cs
using Amazon.JSII.Runtime.Deputy;

namespace Amazon.JSII.Tests.Calculator
{
    [JsiiClass(typeof(SyncVirtualMethods), "jsii-calc.SyncVirtualMethods", "[]")]
    public class SyncVirtualMethods : DeputyBase
    {
        public SyncVirtualMethods(): base(new DeputyProps(new object[]{}))
        {
        }

        protected SyncVirtualMethods(ByRefValue reference): base(reference)
        {
        }

        protected SyncVirtualMethods(DeputyProps props): base(props)
        {
        }

        [JsiiProperty("callerIsProperty", "{\"primitive\":\"number\"}")]
        public virtual double CallerIsProperty
        {
            get => GetInstanceProperty<double>();
            set => SetInstanceProperty(value);
        }

        [JsiiProperty("theProperty", "{\"primitive\":\"string\"}")]
        public virtual string TheProperty
        {
            get => GetInstanceProperty<string>();
            set => SetInstanceProperty(value);
        }

        [JsiiProperty("readonlyProperty", "{\"primitive\":\"string\"}")]
        public virtual string ReadonlyProperty
        {
            get => GetInstanceProperty<string>();
        }

        [JsiiProperty("otherProperty", "{\"primitive\":\"string\"}")]
        public virtual string OtherProperty
        {
            get => GetInstanceProperty<string>();
            set => SetInstanceProperty(value);
        }

        [JsiiProperty("valueOfOtherProperty", "{\"primitive\":\"string\"}")]
        public virtual string ValueOfOtherProperty
        {
            get => GetInstanceProperty<string>();
            set => SetInstanceProperty(value);
        }

        [JsiiProperty("a", "{\"primitive\":\"number\"}")]
        public virtual double A
        {
            get => GetInstanceProperty<double>();
            set => SetInstanceProperty(value);
        }

        [JsiiMethod("callerIsMethod", "{\"primitive\":\"number\"}", "[]")]
        public virtual double CallerIsMethod()
        {
            return InvokeInstanceMethod<double>(new object[]{});
        }

        [JsiiMethod("callerIsAsync", "{\"primitive\":\"number\",\"promise\":true}", "[]")]
        public virtual double CallerIsAsync()
        {
            return InvokeInstanceMethod<double>(new object[]{});
        }

        [JsiiMethod("virtualMethod", "{\"primitive\":\"number\"}", "[{\"name\":\"n\",\"type\":{\"primitive\":\"number\"}}]")]
        public virtual double VirtualMethod(double n)
        {
            return InvokeInstanceMethod<double>(new object[]{n});
        }

        [JsiiMethod("modifyValueOfTheProperty", null, "[{\"name\":\"value\",\"type\":{\"primitive\":\"string\"}}]")]
        public virtual void ModifyValueOfTheProperty(string value)
        {
            InvokeInstanceVoidMethod(new object[]{value});
        }

        [JsiiMethod("retrieveValueOfTheProperty", "{\"primitive\":\"string\"}", "[]")]
        public virtual string RetrieveValueOfTheProperty()
        {
            return InvokeInstanceMethod<string>(new object[]{});
        }

        [JsiiMethod("retrieveReadOnlyProperty", "{\"primitive\":\"string\"}", "[]")]
        public virtual string RetrieveReadOnlyProperty()
        {
            return InvokeInstanceMethod<string>(new object[]{});
        }

        [JsiiMethod("modifyOtherProperty", null, "[{\"name\":\"value\",\"type\":{\"primitive\":\"string\"}}]")]
        public virtual void ModifyOtherProperty(string value)
        {
            InvokeInstanceVoidMethod(new object[]{value});
        }

        [JsiiMethod("retrieveOtherProperty", "{\"primitive\":\"string\"}", "[]")]
        public virtual string RetrieveOtherProperty()
        {
            return InvokeInstanceMethod<string>(new object[]{});
        }

        [JsiiMethod("readA", "{\"primitive\":\"number\"}", "[]")]
        public virtual double ReadA()
        {
            return InvokeInstanceMethod<double>(new object[]{});
        }

        [JsiiMethod("writeA", null, "[{\"name\":\"value\",\"type\":{\"primitive\":\"number\"}}]")]
        public virtual void WriteA(double value)
        {
            InvokeInstanceVoidMethod(new object[]{value});
        }
    }
}